packages/extension-api/src/extension-api.d.ts

```diff
@@ -2424,4 +2424,47 @@ declare module '@podman-desktop/api' {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     export function setValue(key: string, value: any, scope?: 'onboarding'): void;
   }
+
+  export interface CliToolOptions {
+    id: string;
+    name: string;
+    displayName: string;
+    markdownDescription: string;
+    images: ProviderImages;
+    location?: string;
+    version?: string;
+    detection?: {
+      versionOptions: string[]; // run command by name using these options
+      versionParser: (string) => string;
+    };
```

Comment:
I would remove this part for now as I'm not sure if it's the way to move forward or why we need this.

Reply:
I am not sure ether, but the idea is to be able to configure detection without coding, so you can just add it to package.json or call api directly. We can have multiple predefined version parsers that can be referenced by id. Usually version is a sub-command for a cli tool, it also might have options to control output. It could be `--short` that just prints version in a single line (docker-compose) or `-o json` to print json version (kubectl and oc). 

We can create passthrough and json version parser and confiure detection this way in package.json.

for docker-compose

```
detection?: {
    versionOptions: 'version', '--short',
    versionParser: 'passthrough'
}
```

for kubectl 


```
detection?: {
    versionOptions: 'version', '-o', 'json',
    versionParser: 'json',
    versionParserConfig: {
       path: 'clientVersion.gitVersion'
    }
}
```

and so on.